src/components/AgmExtension/ExtensionRequest.vue

```diff
@@ -255,149 +271,206 @@ export default class ExtensionRequest extends Vue {
   onIsFirstAgmChanged (val: boolean): void {
     if (val) {
       this.data.agmYear = this.data.incorporationDate.getFullYear().toString()
+      this.previousAgmDateText = ''
+      this.data.prevAgmDate = null
     } else {
       this.data.agmYear = ''
     }
+    this.checkEligibility()
+  }
+
+  /** Called when isPrevExtension radio group changes. */
+  @Watch('data.isPrevExtension')
+  onIsPrevExtensionChanged (): void {
+    if (!this.data.isPrevExtension) {
+      this.extensionExpiryDateText = ''
+      this.data.prevExpiryDate = null
+    }
+    this.checkEligibility()
   }
 
   /** Called when extension date picker changes. */
   @Watch('extensionExpiryDateText')
   onExtensionDatePickerChanged (val: string): void {
     this.data.prevExpiryDate = val
+    this.checkEligibility()
+  }
+
+  /** Called when previous AGM date picker changes. */
+  @Watch('previousAgmDateText')
+  onPreviousAgmDatePickerChanged (val: string): void {
+    this.data.prevAgmDate = val
+    this.checkEligibility()
   }
 
   /** Called when intended AGM date picker changes. */
   @Watch('intendedAgmDateText')
   onAgmIntendedDatePickerChanged (val: string): void {
     this.data.intendedAgmDate = val
+    this.checkEligibility()
   }
 
-  /** Called when previous AGM date picker changes. */
-  @Watch('previousAgmDateText')
-  onPreviousAgmDatePickerChanged (val: string): void {
-    this.data.prevAgmDate = val
+  /** Called when current date changes. */
+  @Watch('data.currentDate')
+  onCurrentDateChanged (): void {
+    this.checkEligibility()
   }
 
-  @Watch('data.currentDate')
-  @Watch('data.isPrevExtension')
-  @Watch('previousAgmDateText')
-  @Watch('extensionExpiryDateText')
-  onIsPrevExtensionChanged (): void {
-    /* eslint-disable brace-style */
+  /** Eligibility for first agm with no extension requested this year */
+  /* eslint-disable brace-style */
```

Comment:
What's wrong with the brace style, and should you re-enable this check below this  method?

Reply:
Mainly the linter expects the next conditional block to start right away:

```
if () {
} else {
}
```

Which doesn't really allow for comments between them like this:

```
// if block
if () {
}

// else block
else {
}
``` 

Since the logic is a bit tricky to understand I separated each block, but it's mainly a stylistic/readability choice. I could also have the comments inside the conditional blocks but I find it slightly less readable.

But yeah it's probably good to re-enable the brace style afterwards, I'll do that.

Reply:
Sounds good. You may resolve this conversation after re-enabling the eslint rule.